src/UserAction/Generic.php

```diff
@@ -70,28 +77,78 @@ public function init()
      *
      * @param mixed ...$args
      *
+     * @throws Exception
+     *
      * @return mixed
      */
     public function execute(...$args)
     {
-        // todo - assert owner model loaded
-
-        // todo - start transaction, if atomic
-
-        // todo - pass model as first argument ?
-
         // todo - ACL tests must allow
 
-        if ($this->callback === null) {
-            $cb = [$this->owner, substr($this->short_name, strlen('action:'))];
-        } elseif (is_string($this->callback)) {
-            $cb = [$this->owner, $this->callback];
-        } else {
-            array_unshift($args, $this->owner);
-            $cb = $this->callback;
+        try {
+            if (!$this->enabled) {
```

Comment:
This doesn't support `enabled` property as callable. See how it's described above (line 55) https://github.com/atk4/data/pull/440/files#diff-478fc934e86fa5d120af35d752fe76feR55

Reply:
well spotted! Implemented too.